### DISCUSSION TITLE
Retry all endpoints in API using fakeBaseQuery and queryFn

### DISCUSSION THREAD

**Discussion by Janaka-Steph** [reduxjs/redux-toolkit#2132] [Q&A] [answered]
Hello!
I have a big API based on gRPC. Each endpoint is made with `queryFn` and baseQuery is `fakeBaseQuery<string>()`. 
How can I setup `retry` as in https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#automatic-retries ?
Or do I have to check returned data for each call and retry manually?

Thank you

**Comment by phryneas** [comment, marked as answer]
You pretty much have to do that manually I'm afraid - by using `queryFn`, you opt out on everything you do with `baseQuery`.

**Reply by Janaka-Steph** [reply]
Alright, thank you for your reply :)
